f/sliceLazy.js

```diff
@@ -0,0 +1,36 @@
+/**
+ *
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+function* slice(arr, start, end) {
+  if (typeof start !== 'number') start = 0;
+  if (start < 0) start = arr.length + start;
+  if (start < 0) start = 0;
+  if (start >= arr.length) return;
+
+  if (typeof end !== 'number') end = arr.length;
```

Comment:
Sadly, this won’t work as the generator functions are supposed to take generators (of which the length is not known ahead of time).

I see two options: Either we skip the feature of negative end indices or create a ring buffer with the length of length of `-end` which gets filled up and once full yields from the front of the buffer.

@RockoDev WDYT?